### PR TITLE
Fix OAuth clients' realtime updates

### PR DESCRIPTION
**Pull request by taratatach**
Using the `realtime` plugin and `RealTimeQueries` component with `io.cozy-oauth.clients` documents would not work for 2 reasons:
1. the realtime subscriptions would always set documents' doctype to `io.cozy.files`, no matter what data they would receive
2. the `OAuthClientsCollection.destroy()` method would expect cozy-stack to return data while the `DELETE /settings/clients/:id` route only returns a response header when successful

### CODE REVIEW

File: packages/cozy-stack-client/src/OAuthClientsCollection.js [hunk 1]
```diff
@@ -104,17 +104,13 @@ class OAuthClientsCollection extends DocumentCollection {
...
-   * @param {string}                client._id The client's id
-   *
    * @returns {{ data }} The deleted client
    */
-  async destroy({ _id }) {
-    const resp = await this.stackClient.fetchJSON(
-      'DELETE',
-      uri`/settings/clients/${_id}`
-    )
+  async destroy(client) {
```
Comment (paultranvan) [inline packages/cozy-stack-client/src/OAuthClientsCollection.js:109]: nit: `client` is generally referring to a cozy-client instance, I think it would avoid confusion to name the parameter `oauthClient` for instance ;)

File: packages/cozy-client/types/models/permission.d.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export type Document = {
     _type: string;
     type: string;
 };
-export type PermissionVerb = "DELETE" | "ALL" | "GET" | "PATCH" | "POST" | "PUT";
+export type PermissionVerb = "DELETE" | "PUT" | "POST" | "GET" | "ALL" | "PATCH";
```
Comment (nono) [inline packages/cozy-client/types/models/permission.d.ts:55]: Why this change?

File: packages/cozy-client/types/types.d.ts [hunk 3]
```diff
@@ -193,6 +193,27 @@ export type CordovaWindow = {
...
+     */
+    _id?: string;
+    /**
+     * - Current revision of the document
+     */
+    _rev?: string;
+    /**
+     * - When the document has been deleted
+     */
+    _deleted?: string;
```
Comment (nono) [inline packages/cozy-client/types/types.d.ts:211]: Is it a string or a bool?

Comment (taratatach) [reply packages/cozy-client/types/types.d.ts:211]: Good question. I copy/pasted the `CozyClientDocument` type but it's strange to see a `string` type for this attribute.
I'll check it out.

Comment (taratatach) [reply packages/cozy-client/types/types.d.ts:211]: It's definitely a boolean :)
